staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt.go

```diff
@@ -70,9 +73,31 @@ type RequestManagementState struct {
 }
 
 // FlowSchemaSequence holds sorted set of pointers to FlowSchema objects.
-// FlowSchemaSequence implements `sort.Interface` (TODO: implement this).
+// FlowSchemaSequence implements `sort.Interface`
 type FlowSchemaSequence []*rmtypesv1alpha1.FlowSchema
 
+var _ sort.Interface = FlowSchemaSequence{}
+
+func (s FlowSchemaSequence) Len() int {
+	return len(s)
+}
+
+func (s FlowSchemaSequence) Less(i, j int) bool {
+	// the flow-schema w/ lower matching-precedence is prior
+	if ip, jp := s[i].Spec.MatchingPrecedence, s[j].Spec.MatchingPrecedence; ip != jp {
+		return ip < jp
+	}
+	// if the matching-precedence equals, the flow-schema w/ longer/stricter rules is prior
+	if ir, jr := len(s[i].Spec.Rules), len(s[j].Spec.Rules); ir != jr {
```

Comment:
The KEP does not call for this, but also does not forbid this.
The logic here examines only the number of rules, but the comment suggests a more expansive comparison.
I am fine with this as an initial implementation.

Reply:
i updated to alphabetical comparison to make it cleaner, how do you think about this?

Reply:
I am happy with the way it was (apart from the inaccuracy of the comment), I am happy with the way it is.